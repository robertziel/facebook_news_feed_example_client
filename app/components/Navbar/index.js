import React from 'react';

import { NavLink } from 'react-router-dom';
import { Button } from 'components/_ui-elements';
import FontAwesome from 'react-fontawesome';
import LanguageToggle from 'components/LanguageToggle/index';
import SignOutButton from 'components/SignOutButton/index';

import Wrapper from './Wrapper';

function Navbar() {
  return (
    <Wrapper>
      <div className="navbar-right">
        <SignOutButton />
      </div>
      <div className="navbar-right">
        <LanguageToggle />
      </div>
      <div className="navbar-right">
        <Button navbar component={NavLink} exact to="/profile">
          <FontAwesome name="user" />
        </Button>
      </div>
      <div className="navbar-right">
        <Button navbar component={NavLink} exact to="/">
          <FontAwesome name="home" />
        </Button>
      </div>

      <div className="navbar-left">
        <Button navbar component={NavLink} exact to="/new-post">
          <FontAwesome name="plus" />
        </Button>
      </div>
    </Wrapper>
  );
}

export default Navbar;
