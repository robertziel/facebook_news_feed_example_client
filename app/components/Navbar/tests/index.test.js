import React from 'react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';
import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import Navbar from '../index';

// Mock LanguageToggle required by Navbar
/* eslint-disable react/prop-types */
jest.mock('components/LanguageToggle/index', () => () => (
  <div>LanguageToggle</div>
));
/* eslint-enable react/prop-types */

describe('<Navbar />', () => {
  it('should render and match the snapshot', () => {
    const { store } = new ConfigureTestStore();

    const wrapper = mount(
      <IntlProvider locale="en">
        <Provider store={store}>
          <MemoryRouter>
            <Navbar />
          </MemoryRouter>
        </Provider>
      </IntlProvider>,
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
