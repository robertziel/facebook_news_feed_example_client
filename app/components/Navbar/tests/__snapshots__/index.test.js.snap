// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render and match the snapshot 1`] = `"<nav class=\\"Wrapper-rw3tpv-0 bGAmmu\\"><div class=\\"navbar-right\\"><div class=\\"Wrapper-vnjem8-0 gCpCbH\\"><form><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Button-sc-17yegtu-0 gncxZd\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\"><span aria-hidden=\\"true\\" class=\\"fa fa-power-off\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></form></div></div><div class=\\"navbar-right\\"><div>LanguageToggle</div></div><div class=\\"navbar-right\\"><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Button-sc-17yegtu-0 gncxZd\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/profile\\"><span class=\\"MuiButton-label\\"><span aria-hidden=\\"true\\" class=\\"fa fa-user\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></a></div><div class=\\"navbar-right\\"><a aria-current=\\"page\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Button-sc-17yegtu-0 gncxZd active\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/\\"><span class=\\"MuiButton-label\\"><span aria-hidden=\\"true\\" class=\\"fa fa-home\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></a></div></nav>"`;
